src/recipe/__mocks__/recipe.controller.ts

```diff
@@ -0,0 +1,63 @@
+import { faker } from '@faker-js/faker';
+class mockRecipeRequest {
+  title: string;
+  description: string;
+  ingredients: string;
+  preparation: string;
+}
+
+export class mockRecipeController {
+  public createRecipe(dto: mockRecipeRequest) {
+    return Promise.resolve({
+      id: 1,
+      createdAt: Date.now(),
+      ...dto,
+    });
+  }
+
+  public fetchRecipe(id: number) {
+    return Promise.resolve({
+      id,
+      createdAt: Date.now(),
+      title: faker.word.noun(),
+      description: faker.lorem.text(),
+      ingredients: faker.lorem.words(4),
+      preparation: faker.lorem.lines(5),
+    });
+  }
+
+  public fetchAllRecipes() {
+    return Promise.resolve([
+      {
+        id: 0,
+        createdAt: Date.now(),
+        title: faker.word.noun(),
+        description: faker.lorem.text(),
+        ingredients: faker.lorem.words(4),
+        preparation: faker.lorem.lines(5),
+      },
+      {
+        id: 1,
+        createdAt: Date.now(),
+        title: faker.word.noun(),
+        description: faker.lorem.text(),
+        ingredients: faker.lorem.words(4),
+        preparation: faker.lorem.lines(5),
+      },
+    ]);
+  }
+  public updateRecipe(id: number, dto: mockRecipeRequest) {
+    return Promise.resolve({
+      id,
+      createdAt: Date.now(),
+      title: dto.title || faker.word.noun(),
+      description: dto.description || faker.lorem.text(),
+      ingredients: dto.ingredients || faker.lorem.words(4),
+      preparation: dto.preparation || faker.lorem.lines(5),
+    });
```

Comment:
this should use `payload: UpdateRecipeRequest` (though in the controller I would name it `updateRecipeRequest: UpdateRecipeRequest` and do the same here)

If you are worried that calls like `description: dto.description || faker.lorem.text(),` might get too long you can do something like this:

```suggestion
  public updateRecipe(id: number, updateRecipeRequest: UpdateRecipeRequest = {}) {
    const { title, description, ingredients, preparation } = updateRecipeRequest;
    
    return Promise.resolve({
      id,
      createdAt: Date.now(),
      title: title || faker.word.noun(),
      description: description || faker.lorem.text(),
      ingredients: ingredients || faker.lorem.words(4),
      preparation: preparation || faker.lorem.lines(5),
    });
```

Reply:
also you should preferably use `??` instead of `||`. Read [here](https://stackoverflow.com/questions/61480993/when-should-i-use-nullish-coalescing-vs-logical-or) on a reason why.